sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/bitwise.scala

```diff
@@ -28,11 +26,13 @@ import org.apache.spark.sql.types._
  *
  * Code generation inherited from BinaryArithmetic.
  */
-case class BitwiseAnd(left: Expression, right: Expression) extends BinaryArithmetic {
+case class BitwiseAnd(left: Expression, right: Expression)
+  extends BinaryArithmetic with ExpectsInputTypes {
+
   override def symbol: String = "&"
 
-  protected def checkTypesInternal(t: DataType) =
-    TypeUtils.checkForBitwiseExpr(t, "operator " + symbol)
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(TypeCollection(IntegerType, LongType), TypeCollection(IntegerType, LongType))
```

Comment:
do we allow all kind of integral types or only int and long? If input type is byte, the result is also byte, but with your change, we will cast byte to int and result will be int too.